File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Shapes/getshapeinfo.rb
module Asposediagramjava
  module GetShapeInfo
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        shapes = diagram.getPages().getPage(0).getShapes()
        
        i = 0
        while i < shapes.getCount()
            shape = shapes.get(i)
            # Display information about the shapes
            puts "Shape ID : " + shape.getID().to_s
            puts "Name : " + shape.getName().to_s
            puts "Master Shape : " + shape.getMaster().getName().to_s

            i +=1
        end
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithShapes/GetShapeInfo.php
<?php
namespace Aspose\Diagram\WorkingwithShapes;

use com\aspose\diagram\Diagram as Diagram;
class GetShapeInfo{

    public static function run($dataDir=null){

        # Create instance of Diagram
        $diagram = new Diagram($dataDir."Drawing1.vsd");

        $shapes = $diagram->getPages()->getPage(0)->getShapes();

        $i = 0;
        while ($i<(int)(string)$shapes->getCount()) {
            $shape = $shapes->get($i);
            # Display information about the shapes
            print "Shape ID : " . (string)$shape->getID().PHP_EOL;
            print "Name : " . (string)$shape->getName().PHP_EOL;
            print "Master Shape : ".(string)$shape->getMaster()->getName().PHP_EOL;
            $i += 1;
        }
    }
}